pynhm/hydrology/PRMSCanopy.py

```diff
@@ -42,53 +42,48 @@ def __init__(
 
         verbose = True
         super().__init__(
-            storage_type="canopy",
-            id=id,
             control=control,
             params=params,
             verbose=verbose,
         )
 
+        self.name = "PRMSCanopy"
+        # self._current_time = self.control.current_time
+
         # store dependencies
-        self.pkwater_equiv = pkwater_equiv
-        self.transp_on = transp_on
-
-        # define self variables
-        # todo: may need way to initialize interception storage to something non-zero
-        self.intcp_stor_old = np.array(self.nhru * [0.0])
-        self.intcp_stor = np.array(self.nhru * [0.0])
-        # self.tranpiration_on = True  # prms variable Transp_on
-        # self.covden = None  # will be set to covden_sum or covden_win
-        self.interception_form = np.array(self.nhru * [RAIN], dtype=int)
-        self.intcp_transp_on = np.array(self.nhru * [ACTIVE], dtype=int)
-
-        # define information on the output data that will be accessible
-        self.output_data_names = [
-            "intcpstor",
-            "net_rain",
-            "net_snow",
-            "intcp_evap",
-            "rainfall_adj",
-            "snowfall_adj",
-            "potet",
-        ]
-        self.initialize_output_data()
+        self._input_variables_dict = {}
+        self._input_variables_dict["pkwater_equiv"] = variable_factory(
+            pkwater_equiv, "pkwater_equiv"
+        )
+        self._input_variables_dict["transp_on"] = variable_factory(
+            transp_on, "transp_on"
+        )
+        self._input_variables_dict["hru_ppt"] = variable_factory(
+            hru_ppt, "hru_ppt"
+        )
+        self._input_variables_dict["hru_rain"] = variable_factory(
+            hru_rain, "hru_rain"
+        )
+        self._input_variables_dict["hru_snow"] = variable_factory(
+            hru_snow, "hru_snow"
+        )
+        self._input_variables_dict["potet"] = variable_factory(potet, "potet")
 
-        # define information on inflows, outflows, and change in storage
-        # todo: fill these out and implement a residual calculator?
-        self.budget_inflows = [""]
-        self.budget_outflows = [""]
-        self.budget_change_in_storage = [""]
+        # Where does the initial storage come from? Document here.
```

Comment:
this is something we should require in the subclass: being explict about where cold start (or restart) values are found. For GW it's clearly in the parameters, for canopy it is unclear to me. 
We might add a method to StorageUnit that is required to be overriden